src/commands/tileindex-validate/__test__/tileindex.validate.test.ts

```diff
@@ -230,7 +235,7 @@ describe('GridSizeFromString', () => {
   it('should throw error when converting invalid grid size', async () => {
     await assert.rejects(
       GridSizeFromString.from('-1'),
-      new Error('Invalid grid size "-1"; valid values: "10000", "5000", "2000", "1000", "500"'),
+      new Error('Invalid grid size "-1"; valid values: "50000", "10000", "5000", "2000", "1000", "500"'),
```

Comment:
Is it possible to list the values in [gridSizes](https://github.com/linz/argo-tasks/blob/9a7ba359dee9fbd23586ea8a45d72fd6b0c06af7/src/utils/mapsheet.ts#L59)?

Reply:
This is a tricky one, in tests. Basically, how much of the test code should be dynamic? If we generate this expected string, the test doesn't have to change when the list changes. But there are issues with replacing this line with

```typescript
new Error(`Invalid grid size "-1"; valid values: "${gridSizes.join('", "')}"`)
```

or even

```typescript
new Error(`Invalid grid size "${value}"; valid values: "${gridSizes.join('", "')}"`)
```

(the exact [production code](https://github.com/linz/argo-tasks/blob/18576e6a246c46dd65f26dc2a24c5531af9703d6/src/commands/tileindex-validate/tileindex.validate.ts#L111)). It's much harder to tell whether the expected error message is *actually* what we expect. For example, what if we passed the integer literal `1e5` or `100_000` into `GridSizeFromString.from`? The error message would say "100000", not the same as the original literal. Is this what we want? Depends on the application. Maybe we'd want "100,000" in the error message, but only for English, where the thousands separator is usually comma. We're not testing that here, because it's just not a very important use case compared to all the other stuff, but for any code we copy from production we're basically saying we don't care about testing that piece of code. And what if there's a bug in all that mess of nested code and strings? By using a literal as the expected value we're checking that the literal is actually what we want, rather than testing that the same code, executed within the test context and production context produce the same result.

Reply:
Seems too complex. Allgood then!